pkg/nsx/services/subnetport/subnetport.go

```diff
@@ -132,6 +133,15 @@ func (service *SubnetPortService) CreateOrUpdateSubnetPort(obj interface{}, nsxS
 		log.Error(err, "check and update NSX subnet port state failed, would retry exponentially", "nsxSubnetPort.Id", *nsxSubnetPort.Id, "nsxSubnetPath", *nsxSubnet.Path)
 		return nil, err
 	}
+	createdNSXSubnetPort, err := service.NSXClient.PortClient.Get(subnetInfo.OrgID, subnetInfo.ProjectID, subnetInfo.VPCID, subnetInfo.ID, *nsxSubnetPort.Id)
+	if err != nil {
+		log.Error(err, "check and update NSX subnet port failed, would retry exponentially", "nsxSubnetPort.Id", *nsxSubnetPort.Id, "nsxSubnetPath", *nsxSubnet.Path)
+		return nil, err
+	}
+	err = service.SubnetPortStore.Apply(&createdNSXSubnetPort)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Why here we need to get the port via NSX API and apply it into the store again?

Reply:
We need to read allocated IPAddress from the get request.